lib/tasks/test.js

```diff
@@ -67,7 +67,15 @@ module.exports = function(gulp, config) {
 		])
 		.then(function() {
 			if (config.browserTest) {
-				return module.exports.browserTest(gulp, config);
+				return module.exports.browserTest(gulp, config).then(function(){
```

Comment:
I have tried while working in this to do without a then here, however could not make the build fail then. Had to add it in order for it to work, Otherwise the build will always succeeds.

Hard to prove without tests but spent like a day trying to add some for this part and things were getting too tangled so had to give up (mostly due to time pressure and absence of tests for this gulp task. This definitely needs them though, happy to discuss)